File: app/src/main/java/com/slapshotapps/dragonshockey/models/Player.java
package com.slapshotapps.dragonshockey.models;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.Keep;

/**
 * A player object.
 */
@Keep
public class Player implements Parcelable {

    public static final String FORWARD = "F";
    public static final String DEFENSE = "D";
    public static final String GOALIE = "G";
    public static final String PLAYER_NUMBER = "number";

    public String firstName;
    public String lastName;
    public int number;
    public int playerID;
    public String position;
    public String shot;
    public boolean injuredReserved;

    public Player() {
        //default constructor
    }

    public Player(String position) {
        this.position = position;
    }

    public boolean isForward() {
        return FORWARD.equalsIgnoreCase(position);
    }

    public boolean isDefense() {
        return DEFENSE.equalsIgnoreCase(position);
    }

    public boolean isGoalie() {
        return GOALIE.equalsIgnoreCase(position);
    }

    public PlayerPosition getPosition() {
        if (isGoalie()) {
            return PlayerPosition.GOALIE;
        } else if (isDefense()) {
            return PlayerPosition.DEFENSE;
        } else {
            return PlayerPosition.FORWARD;
        }
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.firstName);
        dest.writeString(this.lastName);
        dest.writeInt(this.number);
        dest.writeInt(this.playerID);
        dest.writeString(this.position);
        dest.writeString(this.shot);
        dest.writeByte(this.injuredReserved ? (byte) 1 : (byte) 0);
    }

    protected Player(Parcel in) {
        this.firstName = in.readString();
        this.lastName = in.readString();
        this.number = in.readInt();
        this.playerID = in.readInt();
        this.position = in.readString();
        this.shot = in.readString();
        this.injuredReserved = in.readByte() != 0;
    }

    public static final Parcelable.Creator<Player> CREATOR = new Parcelable.Creator<Player>() {
        @Override
        public Player createFromParcel(Parcel source) {
            return new Player(source);
        }

        @Override
        public Player[] newArray(int size) {
            return new Player[size];
        }
    };
}

File: app/src/main/java/com/slapshotapps/dragonshockey/models/Player.kt
package com.slapshotapps.dragonshockey.models

import android.os.Parcel
import android.os.Parcelable
import androidx.annotation.Keep


/**
 * A player object.
 */
@Keep
class Player : Parcelable {
    var firstName: String? = null

    var lastName: String? = null

    var number: Int = 0

    var playerID: Int = 0

    var position: String? = null

    var shot: String? = null

    var injuredReserved: Boolean = false

    val isForward: Boolean
        get() = FORWARD.equals(position, ignoreCase = true)

    val isDefense: Boolean
        get() = DEFENSE.equals(position, ignoreCase = true)

    val isGoalie: Boolean
        get() = GOALIE.equals(position, ignoreCase = true)

    constructor() {
        //default constructor
    }

    constructor(position: String) {
        this.position = position
    }

    fun getPlayerPosition(): PlayerPosition {
        return if (isGoalie) {
            PlayerPosition.GOALIE
        } else if (isDefense) {
            PlayerPosition.DEFENSE
        } else {
            PlayerPosition.FORWARD
        }
    }

    companion object {
        val FORWARD = "F"

        val DEFENSE = "D"

        val GOALIE = "G"

        val PLAYER_NUMBER = "number"

        @JvmField
        val CREATOR: Parcelable.Creator<Player> = object : Parcelable.Creator<Player> {
            override fun createFromParcel(source: Parcel): Player = Player(source)
            override fun newArray(size: Int): Array<Player?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this() {
        firstName = source.readString()
        lastName = source.readString()
        number = source.readInt()
        playerID = source.readInt()
        position = source.readString()
        shot = source.readString()
        injuredReserved = source.readByte().toInt() != 0
    }

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        dest.writeString(firstName)
        dest.writeString(lastName)
        dest.writeInt(number)
        dest.writeInt(playerID)
        dest.writeString(position)
        dest.writeString(shot)
        dest.writeByte(if (injuredReserved) 1.toByte() else 0.toByte())
    }
}
